httpcore/_utils.py

```diff
@@ -63,3 +64,27 @@ def origin_to_url_string(origin: Origin) -> str:
     scheme, host, explicit_port = origin
     port = f":{explicit_port}" if explicit_port != DEFAULT_PORTS[scheme] else ""
     return f"{scheme.decode('ascii')}://{host.decode('ascii')}{port}"
+
+
+def is_socket_readable(sock_fd: int) -> bool:
+    """
+    Return whether a socket, as identifed by its file descriptor, is readable.
+
+    "A socket is readable" means that it would return immediately with b"" if we
+    called .recv() on it.
```

Comment:
Is this meant to instead read as...

> "A socket is NOT readable" means that it would return immediately with b"" if we called .recv() on it.

?

Reply:
No, "the socket _is_ readable" is the correct (although _very_ intuitive, but that's why it stuck in my head!) way to describe things, at least if we go back to the conversation we've already had about this here…  https://github.com/encode/httpx/pull/143#issuecomment-515181778 (If you look at the comment just above that one, Seth asked the exact same question than the one you're asking now. :-))